src/xdg.rs

```diff
@@ -245,11 +266,14 @@ impl IconLoader {
     fn load(&self, icon: &str, size: u32) -> Result<Icon, Error> {
         let name = icon.into();
 
-        let path = self.icons.get(icon).ok_or(Error::NotFound)?;
-        let path_str = path.to_string_lossy();
+        // Resolve icon from name if it is not an absolute path.
+        let mut path = Path::new(icon);
+        if !path.is_absolute() {
+            path = self.icons.get(icon).ok_or(Error::NotFound)?;
+        }
```

Comment:
Removed the `exists()` check here since both `ImagerReader::open` and `Svg::from_path` call `File::open` internally, which should return `NotFound` anyway when the file doesn't exist.